uint8 pin
string mode
---
uint8 result
